torch/csrc/autograd/profiler.h

```diff
@@ -178,6 +178,9 @@ struct TORCH_API Event final {
 // a std::vector resize from taking a large amount of time inside
 // a profiling  event
 struct RangeEventList {
+  // This mutex is used to serialize access when different threads are writing
+  // to the same instance of RangeEventList.
+  std::mutex mutex_;
```

Comment:
this change is quite serious, if you make RangeEventList thread safe, then I guess there's no point in using thread_local event lists in profiler at all? I guess we should either keep it lock-free (and then update profiler's code to account for that, i.e. we can no long have both thread local and not thread local accesses, e.g. we can have internally two event lists, one is fast (lock-free) and only for thread local access, another is locked and slow, e.g. for cases like async execution that calls end() callbacks on a different thread); Or, alternatively, we could make RangeEventList thread local and simplify code, but then we need to benchmark this carefully to ensure this doesn't cause regression

Reply:
if you could add a small benchmark (e.g. profiling a tight loop with a recorded op inside) and show there's no regression (it might be actually fine since this lock is not contended, assuming most of the time event list is accessed by the same thread), then the current approach should be fine

Reply:
Discussed it a bit more offline and we will add a benchmark to profile the cost of adding a lock.